services/application-mapper/application-question.js

```diff
@@ -1,174 +1,120 @@
+    getColumnName(data);
+
+    // Check to see if id needs custom mapping
+    // Mapping methods sourced from the data dictionary
+    switch (data.id) {
+        // Creates string I,E,S,C,O based on selected options
+        case 'q-applicant-work-details-option':
+            if (Object.keys(oracleJson).find(column => column === columnName)) {
+                const i = Object.keys(oracleJson).findIndex(column => column === columnName);
+                columnValue = Object.values(oracleJson)[i];
+                data.value.forEach(option => {
+                    columnValue = `${columnValue + option[0].toUpperCase()},`;
+                });
+                columnValue = columnValue.slice(0, -1);
+            } else {
+                columnValue = '';
+                data.value.forEach(option => {
+                    columnValue = `${columnValue + option},`;
                 });
-                this.columnValue = this.columnValue.slice(0, -1);
-                break;
+            }
+            break;
+        case 'q-applicant-job-when-crime-happened':
+            if (
+                data.value === true &&
+                Object.keys(oracleJson).find(column => column === columnName)
+            ) {
+                const i = Object.keys(oracleJson).findIndex(column => column === columnName);
+                columnValue = `${Object.values(oracleJson)[i]}I`;
+            } else if (data.value === true) {
+                columnValue = 'I,';
+            }
+            break;
+        case 'q-applicant-did-the-crime-happen-once-or-over-time':
+            if (data.value === 'once') {
+                columnValue = '2';
+            } else if (data.value === 'over a period of time') {
+                columnValue = '3';
+            }
+            break;
+        case 'q-applicant-who-are-you-applying-for':
+            if (data.value === 'myself') {
+                columnValue = 'Y';
+            } else {
+                columnValue = null;
+            }
+            break;
+        case 'q-applicant-physical-injury':
+            columnValue = '';
+            Object.values(data.value).forEach(option => {
+                columnValue = `${columnValue + option}:`;
+            });
+            columnValue = columnValue.slice(0, -1);
+            break;
 
-            // Map Y, N or null
-            case 'q-applicant-infections':
-            case 'q-applicant-pregnancy-loss':
-            case 'q-applicant-affect-future-duration':
-            case 'q-applicant-future-work':
-                if (this.value === 'yes') {
-                    this.columnValue = 'Y';
+        // Maps timestamp to tariff date
+        case 'q--when-was-the-crime-reported-to-police':
+        case 'q-applicant-enter-your-date-of-birth':
+        case 'q-applicant-when-did-the-crime-happen':
+            columnValue = DateTime.fromISO(data.value)
+                .toFormat('dd-MMM-yy')
+                .toLocaleUpperCase();
+            break;
```

Comment:
Similar to the 'Yes' to 'Y' questions, could this be global for all dates to avoid hard coding questions? Or are some dates not formatted?